adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/HttpClientBuilder.java

```diff
@@ -287,16 +289,20 @@ public HttpClient build() {
                 cm = new SingleClientConnManager(registry);
             }
             BasicHttpParams params = new BasicHttpParams();
-            if (socketTimeout > -1)
-            {
+
+            if (proxyHost != null) {
+                params.setParameter(ConnRoutePNames.DEFAULT_PROXY, proxyHost);
```

Comment:
Yes I know - unfortunately they didn't provide an alternative and it works with this version.
I didn't want to upgrade or introduce a new HTTPClient Version for this.